sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerClientTestBase.java

```diff
@@ -60,77 +71,323 @@
+    }
+
+    private static void validateLabeledData(RecognizedForm actualForm, boolean includeTextDetails,
+        List<ReadResult> readResults, DocumentResult documentResult) {
+
+        validateLayoutDataResults(actualForm.getPages(), readResults, null, includeTextDetails);
+        assertEquals(documentResult.getPageRange().get(0), actualForm.getPageRange().getStartPageNumber());
+        assertEquals(documentResult.getPageRange().get(1), actualForm.getPageRange().getEndPageNumber());
+        documentResult.getFields().forEach((label, expectedFieldValue) -> {
+            final FormField<?> actualFormField = actualForm.getFields().get(label);
+            assertEquals(label, actualFormField.getName());
+            assertEquals(expectedFieldValue.getPage(), actualFormField.getPageNumber());
+            if (expectedFieldValue.getConfidence() != null) {
+                assertEquals(expectedFieldValue.getConfidence(), actualFormField.getConfidence());
+            } else {
+                assertEquals(1.0f, actualFormField.getConfidence());
+            }
+            validateFieldValueTransforms(expectedFieldValue, actualFormField, readResults, includeTextDetails);
+        });
+    }
+
+    private static void validateFieldValueTransforms(FieldValue expectedFieldValue, FormField<?> actualFormField,
+        List<ReadResult> readResults, boolean includeTextDetails) {
+        if (expectedFieldValue != null) {
+            if (expectedFieldValue.getBoundingBox() != null) {
+                validateBoundingBoxData(expectedFieldValue.getBoundingBox(),
+                    actualFormField.getValueText().getBoundingBox());
+            }
+            if (includeTextDetails && expectedFieldValue.getElements() != null) {
+                validateReferenceElementsData(expectedFieldValue.getElements(),
+                    actualFormField.getValueText().getTextContent(), readResults);
+            }
+            switch (expectedFieldValue.getType()) {
+                case NUMBER:
+                    assertEquals(expectedFieldValue.getValueNumber(), actualFormField.getFieldValue());
+                    break;
+                case DATE:
+                    assertEquals(expectedFieldValue.getValueDate(), actualFormField.getFieldValue());
+                    break;
+                case TIME:
+                    assertEquals(expectedFieldValue.getValueTime(), actualFormField.getFieldValue());
+                    break;
+                case STRING:
+                    assertEquals(expectedFieldValue.getValueString(), actualFormField.getFieldValue());
+                    break;
+                case INTEGER:
+                    assertEquals(expectedFieldValue.getValueInteger(), actualFormField.getFieldValue());
+                    break;
+                case PHONE_NUMBER:
+                    assertEquals(expectedFieldValue.getValuePhoneNumber(), actualFormField.getFieldValue());
+                    break;
+                case OBJECT:
```

Comment:
empty?

Reply:
ARRAY case : https://github.com/Azure/azure-sdk-for-java/pull/10640/files#diff-a326d1f12f1f93ca97896b082e9ef0e5R374

OBJECT case: https://github.com/Azure/azure-sdk-for-java/pull/10640/files#diff-a326d1f12f1f93ca97896b082e9ef0e5R385

They were needed to be treated differently and so handled here.